controllers/product.js

```diff
@@ -0,0 +1,210 @@
+const path = require('path');
+const fs = require('fs');
+const crypto = require('crypto');
+const { EventEmitter } = require('events');
+const { setImagesList, setVideosList } = require('../services/services');
+
+const eventEmitter = new EventEmitter();
+
+const now = new Date();
+const today = now.toUTCString().slice(5, 25);
+
+eventEmitter.on('fileUploadFailed', () => {
+  const text = `${today} - Error occurred, file upload was failed`;
+
+  fs.readFile('filesUpload.log', function (err, data) {
+    if (err) {
+      console.error(err);
+    }
+
+    let array = [];
+
+    if (data) {
+      array = JSON.parse(data);
+    }
+    array.push(text);
+    const jsonText = JSON.stringify(array, null, 2);
+
+    fs.writeFile('filesUpload.log', jsonText, (err) => {
+      if (err) {
+        console.error(err);
+      }
+      console.log(text);
+    });
+  });
+});
+
+eventEmitter.on('fileUploadStart', () => {
+  const text = `${today} - File upload has started`;
+
+  fs.readFile('filesUpload.log', function (err, data) {
+    if (err) {
+      console.error(err);
+    }
+
+    let array = [];
+
+    if (data) {
+      array = JSON.parse(data);
+    }
+    array.push(text);
+    const jsonText = JSON.stringify(array, null, 2);
+
+    fs.writeFile('filesUpload.log', jsonText, (err) => {
+      if (err) {
+        console.error(err);
+      }
+      console.log(text);
+    });
+  });
+});
+
+eventEmitter.on('fileUploadEnd', (fileName) => {
```

Comment:
https://nodejs.org/en/learn/manipulating-files/writing-files-with-nodejs#appending-content-to-a-file